packages/react-components/react-nav-preview/src/components/NavItem/NavItem.types.ts

```diff
@@ -4,11 +4,15 @@ import { NavItemValue } from '../NavContext.types';
 export type NavItemSlots = {
   root: Slot<'a'>;
 
-  // TODO - light this up when we get design spec
-  // /**
-  //  * Icon that renders before the content.
-  //  */
-  // icon?: Slot<'span'>;
+  /**
+   * Icon that renders before the content when the item is selected.
+   */
+  selectedIcon?: Slot<'span'>;
+
+  /**
+   * Icon that renders before the content when the item is unselected.
+   */
+  unSelectedIcon?: Slot<'span'>;
```

Comment:
I think this is not how we should handle the icon. From what I see, the icon is the same one and we just apply a filled vs regular one. We have a functionality to make this a "single" icon with `bundleIcons`. Doing this, we would then change it to a single slot.

Here's an example of it in button, you would have the type of the icon be like this:
https://github.com/microsoft/fluentui/blob/3905433ffd099d9746011600fb17a19b2d4d18e2/packages/react-components/react-button/src/components/Button/Button.types.ts#L13

Then in your styles you would apply these classnames to show and hide the icon:
https://github.com/microsoft/fluentui/blob/3905433ffd099d9746011600fb17a19b2d4d18e2/packages/react-components/react-button/src/components/ToggleButton/useToggleButtonStyles.styles.ts#L24

This for hover: https://github.com/microsoft/fluentui/blob/3905433ffd099d9746011600fb17a19b2d4d18e2/packages/react-components/react-button/src/components/Button/useButtonStyles.styles.ts#L197

This for Clicked: https://github.com/microsoft/fluentui/blob/3905433ffd099d9746011600fb17a19b2d4d18e2/packages/react-components/react-button/src/components/Button/useButtonStyles.styles.ts#L212

And this is the bundleIcon util that you need to use to get the result you are looking for: https://github.com/microsoft/fluentui/blob/3905433ffd099d9746011600fb17a19b2d4d18e2/packages/react-components/react-button/stories/ToggleButton/ToggleButtonIcon.stories.tsx#L5

```suggestion
  /**
   * Icon that renders before the content when the item.
   */
  icon?: Slot<'span'>;
```

I also think you might need to support the icon position prop(?) though if the design doesn't add it, you can leave it as is.

```ts
  /**
   * Icon position on the Nav Item.
   *
   * @default `before`
   */
  iconPosition?: 'before' | 'after';
```

Sorry I just noticed it 😅 last time I didn't have the whole context when we talked about it offline.

Reply:
Okay - I think I've got this updated. Let me know if you disagree with my slightly different approach.

Design didn't call for an iconPosition prop, so I'm going to leave as is it until someone asks for it.